### PR TITLE
Routes and other modifications with the new dashboard

### LINKED ISSUE #2256: Routes and other modifications with the new dashboard
- username.cartodb.com should redirect to the user dashboard in the maps view if the user is logged in.
- username.cartodb.com should redirect to the public user dashboard in the maps view if the user is not logged in.
- username.cartodb.com should redirect to the public user dashboard in the maps view if the username is not the user's username.

### PR DESCRIPTION
**Pull request by Kartones**
Fixes #2256

### CODE REVIEW

**Comment by Kartones** [comment]
@rafatower CR please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2070/))

File: app/controllers/admin/pages_controller.rb [hunk 1]
```diff
@@ -34,7 +48,7 @@ def sitemap
       # Redirect to org url if has only user
       if viewed_user.has_organization?
         if CartoDB.extract_real_subdomain(request) != viewed_user.organization.name
-          redirect_to CartoDB.base_url(viewed_user.organization.name) <<  public_sitemap_pathand and return
```
Comment (rafatower) [inline app/controllers/admin/pages_controller.rb]: `_pathand`, was this a typo?

**Comment by rafatower** [comment]
ok

File: app/controllers/admin/pages_controller.rb (hunk 1 above)
Comment (Kartones) [inline app/controllers/admin/pages_controller.rb]: yep!

**Pull request merged**